stdlib/result.mli

```diff
@@ -0,0 +1,89 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Result values.
+
+    @since 4.08 *)
+
+(** {1:results Results} *)
+
+type ('a, 'e) t = ('a, 'e) result = Ok of 'a | Error of 'e
+(** The type for result values. Either a value [Ok v] or an error [Error e]. *)
+
+val ok : 'a -> ('a, 'e) result
+(** [ok v] is [Ok v]. *)
```

Comment:
I personally find this documentation string confusing.

Reply:
How about `[ok v] returns the value [Ok v] for any v.`

Reply:
It's about emphasizing equalities, and not computational behaviour, of pure functions.

Think of that "is" as the function `(=)`.

Reply:
See below.

Reply:
I understand, but documentation is meant to be explanatory, not minimal. The explanation is correct, but you want to help beginners, not just experts.

Reply:
Sorry I'm not too obsessed with beginners, there are introductory texts and tutorials for that. I'm obsessed with the working OCaml programmer.

Reply:
Beginners will never become working OCaml programmers if they hit too many obstacles. A word or two here and there hardly slows down the experienced programmer, but will make it easier for people who are new. Note that I've been using the language for half a year and this documentation string stopped me cold for a minute until I read the code itself. That might not be what you want, correct?